File: hao-ran-music-frontend/src/api/userWork.ts
import { request } from '@/utils/request';
export type WorkType = number;
export type WorkStatus = number;
export type WorkLanguage = number;
export interface UserWork {
    id?: string | number;
    userId: string;
    nickname?: string;
    avatar?: string;
    workType: WorkType;
    workName: string;
    coverUrl: string;
    fileUrl: string;
    fileUrls?: string | string[];
    zipFileUrl?: string;
    lyricFileUrl?: string;
    uploadType?: number;
    versionType?: string;
    productionType?: string;
    fileSize?: number;
    duration?: number;
    description: string;
    tags?: string;
    language: WorkLanguage;
    lyric?: string;
    showRealName?: boolean;
    source?: number;
    status: WorkStatus;
    reviewerId?: string | number;
    reviewerName?: string;
    reviewTime?: string;
    reviewReason?: string;
    publishTime?: string;
    songId?: string | number;
    playCount?: number;
    likeCount?: number;
    collectCount?: number;
    shareCount?: number;
    downloadCount?: number;
    virusScanned?: boolean;
    virusScanResult?: string;
    rewardPoints?: number;
    createTime: string;
    updateTime?: string;
    allowDownload?: number;
    allowComment?: number;
    allowShare?: number;
}
export interface WorkStats {
    totalCount: number;
    pendingCount: number;
    publishedCount: number;
    rejectedCount: number;
}
export interface RewardLevel {
    name: string;
    points: number;
    likeRequired: number;
    collectRequired: number;
    playRequired: number;
}
export interface RewardProgress {
    currentLevel: string;
    nextLevel?: string;
    currentPoints: number;
    nextLevelPoints?: number;
    likeCount: number;
    collectCount: number;
    playCount: number;
    progress: {
        like: number;
        collect: number;
        play: number;
    };
    completed: {
        like: boolean;
        collect: boolean;
        play: boolean;
    };
}
export interface RewardProgressResponse {
    workId: string;
    workName: string;
    status: number;
    rewardPoints: number;
    progress: RewardProgress;
}
export interface RewardRules {
    levels: {
        bronze: RewardLevel;
        silver: RewardLevel;
        gold: RewardLevel;
        platinum: RewardLevel;
    };
    description: string;
    note: string;
}
export function submitWork(workData: Partial<UserWork>) {
    return request<number>({
        url: '/user-work/submit',
        method: 'POST',
        data: workData
    });
}
export function getMyWorks(status?: WorkStatus) {
    return request<UserWork[]>({
        url: '/user-work/my',
        method: 'GET',
        params: { status }
    });
}
export function getWorkDetail(workId: string | number) {
    return request<UserWork>({
        url: `/user-work/${workId}`,
        method: 'GET'
    });
}
export function updateWork(workId: string | number, workData: Partial<UserWork>) {
    return request<boolean>({
        url: `/user-work/${workId}`,
        method: 'PUT',
        data: workData
    });
}
export function deleteWork(workId: string | number) {
    return request<boolean>({
        url: `/user-work/${workId}`,
        method: 'DELETE'
    });
}
export function getWorkStats() {
    return request<WorkStats>({
        url: '/user-work/stats',
        method: 'GET'
    });
}
export function getRewardProgress(workId: string | number) {
    return request<RewardProgressResponse>({
        url: `/user-work/${workId}/reward-progress`,
        method: 'GET'
    });
}
export function getRewardRules() {
    return request<RewardRules>({
        url: '/user-work/reward-rules',
        method: 'GET'
    });
}
export const userWorkApi = {
    submitWork,
    getMyWorks,
    getWorkDetail,
    updateWork,
    deleteWork,
    getWorkStats,
    getRewardProgress,
    getRewardRules
};

File: hao-ran-music-backend/src/main/java/com/haoran/music/entity/UserWork.java
package com.haoran.music.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;





@Data
@TableName("user_work")
public class UserWork implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;




    private Long userId;




    private String nickname;




    private String avatar;




    private Integer workType;




    private String workName;




    private String coverUrl;




    private String fileUrl;




    private String fileUrls;




    private String zipFileUrl;




    private String lyricFileUrl;




    private Integer uploadType;




    private Integer qualityType;




    private String versionType;




    private String productionType;




    private Long fileSize;




    private Integer duration;




    private String description;




    private String tags;




    private Integer language;




    private String lyric;




    private Boolean showRealName;




    private Integer source;




    private Integer status;




    private Long reviewerId;




    private String reviewerName;




    private LocalDateTime reviewTime;




    private String reviewReason;




    private LocalDateTime publishTime;




    private Long songId;




    @TableField(exist = false)
    private Long albumId;




    @TableField(exist = false)
    private String albumName;




    private Long playCount;




    private Integer likeCount;




    private Integer collectCount;




    private Integer shareCount;




    private Integer downloadCount;




    private Boolean virusScanned;




    private String virusScanResult;




    private Integer rewardPoints;






    private Integer allowDownload;




    private Integer allowComment;




    private Integer allowShare;




    private LocalDateTime createTime;




    private LocalDateTime updateTime;




    @TableLogic
    private Integer deleted;
}
